main.nf

```diff
@@ -249,46 +249,43 @@ process CheckSampleSheet {
     """
 }
 
-// Function to see if fasta file exists in iGenomes
-def get_fasta(genome, genomeMap) {
-    def fasta = null
-    if (genome) {
-        if (genomeMap.containsKey(genome)) {
-            fasta = file(genomeMap[genome].fasta, checkIfExists: true)
+// Function to resolve fasta and gtf file if using iGenomes
+// Returns [ sample, fastq, barcode, fasta, gtf, is_transcripts ]
+def get_sample_info(LinkedHashMap sample, LinkedHashMap genomeMap) {
+
+    // Resolve fasta and gtf file if using iGenomes
+    def fasta = false
+    def gtf = false
+    if (sample.genome) {
+        if (genomeMap.containsKey(sample.genome)) {
+            fasta = file(genomeMap[sample.genome].fasta, checkIfExists: true)
+            gtf = file(genomeMap[sample.genome].gtf, checkIfExists: true)
```

Comment:
When using an iGenomes file, we could save the `gtf2bed` conversion if we just pull either the `bed12` or `gtf` depending on `params.aligner`.
Also are we therefore performing transcript aware genome alignment by default? There might be contexts in which we don't want this by default - I see you have the `DNA` case accounted for in the alignment processes but this might need to be an option for RNAseq rather than default, even if it does improve alignment quality
It might be good if we require the entry in the `transcriptome` column too to pull the `gtf`.
I.e.
```
sample, fastq, barcode, genome, transcriptome
sample1,,,hg19,hg19
sample2,,,hg19,
```
Where `sample1` would undergo transcript aware genome alignment and `sample2` would undergo standard genome alignment.

Reply:
The complications isnt pulling the `bed12` or `gtf` file, its actually staging/storing all of the files together properly per sample. I think its going to be more hassle then its worth to implement pulling the `bed12` too rather then just converting it within the pipeline. Happy for you to have a go at adding the DNA logic in a separate PR?

Reply:
Yeah after looking at all your channel magic I think having the `bed12` will indeed be a lot of hassle! 
I mean if given an iGenomes entry, it automatically pulls the `gtf` & the `genome.fa` and then performs the alignment with that (unless it's DNA). But if there was no entry in the `transcriptome` section of `samplesheet.csv` then it should just be standard genome alignment. 
Can look at this in a later PR unless I misread and this is already happening 😅

Reply:
Nice :+1: